File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string myString, string pat) {
    int answer = 0;
    for(int i = 0; i < myString.size(); i++){
        if(myString[i] == 'A') myString[i] = 'B';
        else myString[i] = 'A';
    }
    
    return myString.find(pat) != string::npos;
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        char[] arr = myString.toCharArray();

        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == 'A') arr[i] = 'B';
            else arr[i] = 'A';
        }

        String result = new String(arr);
        return result.contains(pat) ? 1 : 0;
    }
}